doc/api/http.md

```diff
@@ -1520,15 +1537,13 @@ added: v0.3.6
   * `headers` {Object} An object containing request headers.
   * `auth` {String} Basic authentication i.e. `'user:password'` to compute an
     Authorization header.
-  * `agent` {http.Agent|Boolean} Controls [`Agent`][] behavior. When an Agent
-    is used request will default to `Connection: keep-alive`. Possible values:
+  * `agent` {http.Agent|Boolean} Controls [`Agent`][] behavior. Possible values:
    * `undefined` (default): use [`http.globalAgent`][] for this host and port.
    * `Agent` object: explicitly use the passed in `Agent`.
-   * `false`: opts out of connection pooling with an Agent, defaults request to
-     `Connection: close`.
+   * `false`: causes a new `Agent` with default values to be used.
```

Comment:
@lance The socket is only reused if `'free'` is emitted as you can see in your link.

Here's the entire client request flow:

1. [`http.request()` merely creates a new `ClientRequest()`](https://github.com/nodejs/node/blob/accf410eb0992628b4aab3c139242a409874a4fb/lib/http.js#L26)
2. [`ClientRequest()` creates a new `Agent` instance using the default options when `agent: false` is passed](https://github.com/nodejs/node/blob/fc7025c9c011f266c6bd73e1fb634eb5f519c5cb/lib/_http_client.js#L61-L64)
  a. [As we see in the `Agent()` constructor, the default value for `keepAlive` is `false`](https://github.com/nodejs/node/blob/c04d4df08c0ba10d4402b09796233fb4f7b52c6c/lib/_http_agent.js#L39)
  b. [We also see that in the `Agent()` constructor that the default value for `maxSockets` is `Agent.defaultMaxSockets`](https://github.com/nodejs/node/blob/c04d4df08c0ba10d4402b09796233fb4f7b52c6c/lib/_http_agent.js#L40) which [is `Infinity`](https://github.com/nodejs/node/blob/c04d4df08c0ba10d4402b09796233fb4f7b52c6c/lib/_http_agent.js#L89).
3. [Later in `ClientRequest()`, we check if the agent is using `keepAlive` and if the `maxSockets` is set to `Infinity`. If so, then we disable reuse of sockets by setting `req.shouldKeepAlive` to `false`.](https://github.com/nodejs/node/blob/fc7025c9c011f266c6bd73e1fb634eb5f519c5cb/lib/_http_client.js#L211-L218)
  a. [As a side note, if the server is sending `Connection: close` or something similar, then we definitely do not reuse the socket, even if `agent.keepAlive` is `true`.](https://github.com/nodejs/node/blob/fc7025c9c011f266c6bd73e1fb634eb5f519c5cb/lib/_http_client.js#L519-L524)
4. [When we go to send the initial HTTP request line and headers, we see that if the user has not explicitly set a `Connection` header, that the default behavior is to check `req.shouldKeepAlive` (among other values) to determine whether to send `Connection: keep-alive` or not. Importantly, if `req.shouldKeepAlive` is `false`, then `Connection: close` is sent.](https://github.com/nodejs/node/blob/dab22b5b7ab61c59997644fb699968a632407f72/lib/_http_outgoing.js#L276-L283)
5. [When the response from the server is complete, `req.shouldKeepAlive` is checked and `responseKeepAlive()` is called because `req.shouldKeepAlive` is `false`.](https://github.com/nodejs/node/blob/fc7025c9c011f266c6bd73e1fb634eb5f519c5cb/lib/_http_client.js#L577-L578)
6. [Once inside `responseKeepAlive()`, we immediately check if `req.shouldKeepAlive` is `false`. If so, we destroy the socket if it hasn't already been destroyed.](https://github.com/nodejs/node/blob/fc7025c9c011f266c6bd73e1fb634eb5f519c5cb/lib/_http_client.js#L551-L556). You can see if the `else` block that when `req.shouldKeepAlive` is `true`, that we emit `'free'`, which leads to the code you pointed out executing.